### PR TITLE
feat: Add ability to modify headers in the tree

### LINKED ISSUE #903: Manipulation of markdown by header level.
## Issue description / feature objectives
Presume I have an existing piece of markdown at header level 2 (e.g. `##`), however, I am injecting this into a document where the "containing header" is already at level 3 (e.g. `###`). For this scenario if I want this I would want a proportional increase in the header such it fits within the existing header (e.g. `##` -> `####` and `###` -> `#####`).

This implies two phases at a high level.
1) calculate the header level for a block
2) Adjust the header level.

This could either be performed using the mdnode api or similar.

Second phase would be to create a jinja flag that could do the same.

### PR DESCRIPTION
**Pull request by enikonovad**
Signed-off-by: Ekaterina Nikonova <enikonovad@gmail.com>

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
Added a functionality to modify the headers inside of the `MarkdownNode` by the given value. Also added a function to get the level of the node header.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar

Closes #903

### CODE REVIEW

File: trestle/core/markdown/markdown_node.py [hunk 1]
```diff
@@ -158,6 +188,29 @@ def _build_tree(
         md_node.subnodes = node_children
         return (md_node, i)
 
+    def _modify_header_level(self, header: str, level: int) -> str:
+        """Modify header level by specified level."""
+        if level == 0:
```
Comment (fsuits) [inline trestle/core/markdown/markdown_node.py:193]: I would call this "delta_level" or just delta to emphasize it represents a change of level and not a level.

**Review by fsuits — APPROVED** [review]
LGTM with minor suggestion

File: trestle/core/markdown/markdown_node.py (hunk 1 above)
Comment (enikonovad) [reply trestle/core/markdown/markdown_node.py]: Good suggestion, I think this name will add more clarity. Added

**Review by butler54 — APPROVED** [review]
LGTM

**Pull request merged**